### ISSUE TITLE
New additional parameter $defaultTenantHasAllowAppDatabaseWrite does not seems to work in case of newdatabase switch

### ISSUE THREAD

**Issue by ekeukens** [microsoft/navcontainerhelper#1414]
This new additional parameter only seems to be implemented for switching the default database to multi-tenant.
In case of new/clean database this parameter doesn't affect the setting of the default tenant.

See two runs: first one is switching the default database to multi tenat. The second one is using cleandatabase:
```
New-BCContainer -accept_eula
-containerName $ContainerName 
-Auth NavUserPassword
-Credential $credential 
-artifactUrl $ArtifactUrl
-imageName $ImageName 
-licensefile $LocalLicenseFileName
-updateHosts 
-myScripts @("$LocalCollationFileName")
-memoryLimit 20G 
-multitenant
-TimeZoneId (Get-TimeZone).id 
-copyTables @('Add-in', 'Report Layout', 'Style Sheet')
-additionalParameters @("--env defaultTenantHasAllowAppDatabaseWrite=Y", 
                        "--volume ${LocalHostSharePath}:${ContainerHostSharePath}",
                        "--volume ${LocalAddInsPath}:${ContainerAddInsPath}")
NavContainerHelper is version 0.7.0.26
NavContainerHelper is running as administrator
Host is Microsoft Windows 10 Pro - 1903
Docker Client Version is 19.03.13
Docker Server Version is 19.03.13
Removing container bc165
Removing bc165 from host hosts file
Removing C:\ProgramData\NavContainerHelper\Extensions\bc165
ArtifactUrl and ImageName specified
Fetching all docker images
Image myown_16.5.15897.17019:sandbox-16.5.15897.17019-nl already exists
Using image myown_16.5.15897.17019:sandbox-16.5.15897.17019-nl
Creating Container bc165
Version: 16.5.15897.17019-NL
Style: sandbox
Platform: 16.0.15884.16928
Generic Tag: 0.1.0.24
Container OS Version: 10.0.18362.1016 (1903)
Host OS Version: 10.0.18362.1016 (1903)
Using locale nl-NL
Using process isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\Users\ekeukens\Documents\AL\FpsFameFall\Scripts\BcContainer\4805272.flf
Additional Parameters:
--env defaultTenantHasAllowAppDatabaseWrite=Y
--volume C:\bacpac165:C:\hostshare
--volume C:\bacpac165\Add-Ins:C:\Run\Add-Ins
Files in C:\ProgramData\NavContainerHelper\Extensions\bc165\my:

AdditionalOutput.ps1
Collation.txt
license.flf
MainLoop.ps1
SetupVariables.ps1
updatehosts.ps1
Creating container bc165 from image myown_16.5.15897.17019:sandbox-16.5.15897.17019-nl
e9bad388958a60d9937a6fa96be93cce45b0642da4a05404427a0ac52b456f1a
Waiting for container bc165 to be ready
Initializing...
Setting host.docker.internal to 192.168.178.15 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.178.15 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.20.96.1 in container hosts file
Starting Container
Hostname is bc165
PublicDnsName is bc165
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Copying Database on localhost\SQLEXPRESS from CRONUS to tenant
Taking database CRONUS offline
Copying database files
Attaching files as new Database tenant
Putting database CRONUS back online
Removing Database CRONUS from localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Creating Self Signed Certificate
Self Signed Certificate Thumbprint F8BC0C8A18F8F085A34E4F721934EE2E73E54145
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Enabling rewrite rule: Don't rewrite system files
Enabling rewrite rule: Already have tenant specified
Enabling rewrite rule: Hostname (without port) to tenant
Using license file 'c:\run\my\license.flf'
Import License
Copying template database
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting Tenant
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = True
Sync'ing Tenant
Tenant is Operational
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
Creating SUPER user
Container IP Address: 172.20.107.202
Container Hostname : bc165
Container Dns Name : bc165
Web Client : http://bc165/BC/?tenant=default
Dev. Server : http://bc165
Dev. ServerInstance : BC
Dev. Server Tenant : default
Setting bc165 to 172.20.107.202 in host hosts file
Files:
http://bc165:8080/ALLanguage.vsix

WARNING: You are running a container which is 76 days old.
Microsoft recommends that you always run the latest version of our containers.

Container Total Physical Memory is 31.9Gb
Container Free Physical Memory is 12.9Gb

Initialization took 163 seconds
Ready for connections!
Reading CustomSettings.config from bc165
Set TimeZone in Container to W. Europe Standard Time
Creating Desktop Shortcuts for bc165
Container bc165 successfully created




New-BCContainer 
-accept_eula
-containerName $ContainerName 
-Auth NavUserPassword
-Credential $credential 
-artifactUrl $ArtifactUrl
-imageName $ImageName 
-licensefile $LocalLicenseFileName
-updateHosts 
-myScripts @("$LocalCollationFileName")
-memoryLimit 20G 
-multitenant
-UsecleanDatabase 
-useNewDatabase
-TimeZoneId (Get-TimeZone).id 
-copyTables @('Add-in', 'Report Layout', 'Style Sheet')
-additionalParameters @("--env defaultTenantHasAllowAppDatabaseWrite=Y", 
                        "--volume ${LocalHostSharePath}:${ContainerHostSharePath}",
                        "--volume ${LocalAddInsPath}:${ContainerAddInsPath}")
NavContainerHelper is version 0.7.0.26
NavContainerHelper is running as administrator
Host is Microsoft Windows 10 Pro - 1903
Docker Client Version is 19.03.13
Docker Server Version is 19.03.13
Removing container bc165
Removing bc165 from host hosts file
Removing C:\ProgramData\NavContainerHelper\Extensions\bc165
ArtifactUrl and ImageName specified
Fetching all docker images
Image myown_16.5.15897.17019:sandbox-16.5.15897.17019-nl already exists
Using image myown_16.5.15897.17019:sandbox-16.5.15897.17019-nl
Creating Container bc165
Version: 16.5.15897.17019-NL
Style: sandbox
Platform: 16.0.15884.16928
Generic Tag: 0.1.0.24
Container OS Version: 10.0.18362.1016 (1903)
Host OS Version: 10.0.18362.1016 (1903)
Using locale nl-NL
Using process isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\Users\ekeukens\Documents\AL\FpsFameFall\Scripts\BcContainer\4805272.flf
Additional Parameters:
--env defaultTenantHasAllowAppDatabaseWrite=Y
--volume C:\bacpac165:C:\hostshare
--volume C:\bacpac165\Add-Ins:C:\Run\Add-Ins
Files in C:\ProgramData\NavContainerHelper\Extensions\bc165\my:

AdditionalOutput.ps1
Collation.txt
license.flf
MainLoop.ps1
SetupVariables.ps1
updatehosts.ps1
Creating container bc165 from image myown_16.5.15897.17019:sandbox-16.5.15897.17019-nl
829fd9d4d966903d955db55db9a3c87e921f434114f19311a21fca8068fa767b
Waiting for container bc165 to be ready
Initializing...
Setting host.docker.internal to 192.168.178.15 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.178.15 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.20.96.1 in container hosts file
Starting Container
Hostname is bc165
PublicDnsName is bc165
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 02A6E4860382E96786F59FAC54C457ABC266769F
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
Creating SUPER user
Container IP Address: 172.20.99.200
Container Hostname : bc165
Container Dns Name : bc165
Web Client : http://bc165/BC/
Dev. Server : http://bc165
Dev. ServerInstance : BC
Setting bc165 to 172.20.99.200 in host hosts file
Files:
http://bc165:8080/ALLanguage.vsix

WARNING: You are running a container which is 76 days old.
Microsoft recommends that you always run the latest version of our containers.

Container Total Physical Memory is 31.9Gb
Container Free Physical Memory is 13.6Gb

Initialization took 77 seconds
Ready for connections!
Reading CustomSettings.config from bc165
Set TimeZone in Container to W. Europe Standard Time
Creating Desktop Shortcuts for bc165
Stopping ServiceTier in order to replace database
Copying Database on localhost\SQLEXPRESS from CRONUS to mytempdb
Taking database CRONUS offline
Copying database files
Attaching files as new Database mytempdb
Putting database CRONUS back online
Removing Database CRONUS from localhost\SQLEXPRESS
Creating new database CRONUS on localhost\SQLEXPRESS with Collation Latin1_General_100_CI_AS
Copying table [Add-in] from original database
Copying table [Report Layout] from original database
Copying table [Style Sheet] from original database
Copying table [Entitlement] from original database
Copying table [Entitlement Set] from original database
Copying table [Membership Entitlement] from original database
Removing Database mytempdb from localhost\SQLEXPRESS
Starting Service Tier
Synchronizing
Importing license file
Importing License C:\ProgramData\NavContainerHelper\Extensions\bc165\my\license.flf
Creating user admin
Creating User admin
Assigning Permission Set SUPER to admin
Publishing System Symbols
Copy from container bc165 (C:\Program Files\Microsoft Dynamics NAV\160\AL Development Environment\System.app) to C:\ProgramData\NavContainerHelper\Extensions\bc165_System.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\bc165_System.app
App successfully published
Creating Company
Creating company My Company in default
Company successfully created
Publishing System Application
Copy from container bc165 (C:\Applications\System Application\Source\Microsoft_System Application.app) to C:\ProgramData\NavContainerHelper\Extensions\bc165_Microsoft_System Application.app
Publishing C:\ProgramData\NavContainerHelper\Extensions\bc165_Microsoft_System Application.app
Synchronizing System Application on tenant default
Installing System Application on tenant default
App successfully published
Switching to multitenant
Copying Database on localhost\SQLEXPRESS from CRONUS to tenant
Taking database CRONUS offline
Copying database files
Attaching files as new Database tenant
Putting database CRONUS back online
Removing Database CRONUS from localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
WARNING: The new settings value will not take effect until you stop and restart the service.
Creating Tenant default on bc165
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant successfully created
Container bc165 successfully created

**Comment by freddydk** [comment]
True. As the default tenant in this case is created inside new-bccontainer that doesn't get this.
I will fix this in the next version of BcContainerHelper

**Comment by freddydk** [comment]
Shipped in BcContainerHelper 1.0.13-preview256
